### ISSUE TITLE
Bundles class name conflict with Assets

### ISSUE THREAD

**Issue by lclement57** [T4MVC/T4MVC#18]
I am using a library which has a folder named `assets`

With the version 3.14.1, the code generated by the method `ProcessBundleFilesRecursive` creates a partial class for this folder named `assets` aside the `Assets` class.

It compiles, but raises a warning for the class nont being CLS-Compliant since the two classes differ only by case.
If the folder was named `Assets`, then it would be merged with the list of scripts in the parent Directory

Maybe the term `Assets` should be added to the list of reserved keywords when sanitizing the project item name `ProcessBundleFilesRecursive` ?

**Comment by AdrianLogan** [comment]
I also have a folder called assets but get a different error within bundles. 'Assets': member names cannot be the same as their enclosing type". This results in a compiler error. Worked fine before the upgrade. How can I switch this off as we do not use the MVC bundles?

```
public static partial class Bundles
{
        public static partial class Assets 
        { 
            ... all the classes with the links. But at the end we get
            public static class Assets     // <---- This empty class is causing the error
            {
            }
}
```

**Comment by davidebbo** [comment]
@kevinkuszyk this relates to your change c1dbc6fcbc8d60b1aaf28c3d314760a2935dcae9. Can you help work through a solution/workaround for it? Thanks!

**Comment by kevinkuszyk** [comment]
I'll try to have a look at it tomorrow.

**Comment by kevinkuszyk** [comment]
Thanks @lclement57 and @AdrianLogan.  I believe I have reproduced your issues in my [fork](https://github.com/kevinkuszyk/T4MVC/tree/asset-collision).  Can you confirm I have understood your scenarios?

I think we have three options:
1. Add `assets` and `Assets` to the list of reserved words.  If we find folders with these names, they will be generated as `_assets` and `_Assets`.
2. Add a setting to override the the name of the `Assets` class we generate.
3. Add a setting to switch off generating the asset name constants for bundles. 

My preference is option 2, but I'm happy to go with the consensuses.

I remember we had quite a bit of discussion about what to call these constants on [CodePlex](https://t4mvc.codeplex.com/discussions/442706).  It sounds like the design is not quite right yet.

Thoughts?

**Comment by lclement57** [comment]
My scenario a little bit different, but its the same issue:
I have my JS library under the Scripts folder that contains a subfolder named assets, like: `Scripts\myLib\assets`

So the generated code looks like

```
public static partial class Bundles
{
    public static partial class Scripts 
    {
        public static partial class myLib 
        {
            public static partial class assets 
            {
                public static class Assets
                {
                    public const string ...
                }
            }
            public static class Assets
            {
            }
        }
    }
}
```

I'd prefer option 1 for this scenario, since with option 2 we could still choose a name that a library might be using, but with the underscore at the end like it's already done when two folders have the same name in the hierarchy.

The generated code for a structure like `Scripts\folder\folder` is currently

```
public static partial class Bundles
{
    public static partial class Scripts 
    {
        public static partial class folder 
        {
            public static partial class folder_
        }
    }
}
```

**Comment by lclement57** [comment]
The issue is still open, so here is the modification I suggest to handle the name conflict and the casing:
In the method `void ProcessBundleFilesRecursive(ProjectItem projectItem, string path, HashSet<String> nameSet)`, change the `childrenNameSet` HashSet to be case insenstitive and reserve the word `Assets`

Replace

```
var childrenNameSet = new HashSet<String>();
```

by

```
var childrenNameSet = new HashSet<String>(StringComparer.OrdinalIgnoreCase);
childrenNameSet.Add("Assets");
```

**Comment by kevinkuszyk** [comment]
Thanks @lclement57, can you send a PR with that change in?

**Comment by davidebbo** [comment]
Fixed by 9402bee6d3fc4c035e58cf57a17b78783a54a11f
